lib/Common/Memory/RecyclerPointers.h

```diff
@@ -58,8 +59,23 @@ struct AllocatorWriteBarrierPolicy
         typename _AllocatorTypeWriteBarrierPolicy<AllocatorType>::Policy,
         typename TypeWriteBarrierPolicy<T>::Policy>::Policy Policy;
 };
+
+template <class T>
+struct AllocatorWriteBarrierPolicy<RecyclerWriteBarrierAllocator, T> { typedef _write_barrier_policy Policy; };
+
+template <>
+struct AllocatorWriteBarrierPolicy<RecyclerWriteBarrierAllocator, int> { typedef _no_write_barrier_policy Policy; };
+
 template <class T>
-struct AllocatorWriteBarrierPolicy<RecyclerNonLeafAllocator, T> { typedef _write_barrier_policy Policy; };
+struct AllocatorWriteBarrierPolicy<RecyclerNonLeafAllocator, T> 
+{
+#if GLOBAL_FORCE_USE_WRITE_BARRIER
+    typedef _write_barrier_policy Policy;
+#else
+    typedef _no_write_barrier_policy Policy; 
+#endif
```

Comment:
As offline conversation, a cleaner way might be introducing a 3rd policy variable -- default global policy, which is on by default xplat; off by default Windows,on if globally force. The effective policy would be determined by: allocator type, data type, default global policy (I could work on that after my current annotation batch.)